### PR TITLE
feat(client): ts-migrate layoutSelector component

### PR DESCRIPTION
**Pull request by sharmaabhinav**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/utils/gatsby/layout-selector.tsx [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
   CertificationLayout,
   DefaultLayout
 } from '../../src/components/layouts';
 import FourOhFourPage from '../../src/pages/404';
 import { isChallenge } from '../../src/utils/path-parsers';
 
-export default function layoutSelector({ element, props }) {
+type Location = {
+  pathname: string;
+};
```
Comment (ojeytonwilliams) [inline client/utils/gatsby/layout-selector.tsx:11]:
```suggestion
interface Location {
  pathname: string;
}
```
Nitpick, we prefer interfaces over type aliases: https://contribute.freecodecamp.org/#/codebase-best-practices?id=interfaces-and-types

File: client/utils/gatsby/layout-selector.tsx [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+
+type LayoutSelectorProps = {
+  props: {
+    location: Location;
+  };
+  element: React.ReactElement;
+};
```
Comment (ojeytonwilliams) [inline client/utils/gatsby/layout-selector.tsx:18]:
```suggestion
interface LayoutSelectorProps = {
  props: {
    location: Location;
  };
  element: React.ReactElement;
}
```
Same as above.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for this PR, @sharmaabhinav.  Tiny nitpicks aside, this looks good.

**Review by ojeytonwilliams — APPROVED** [review]
Perfect, thanks @sharmaabhinav. LGTM :+1:

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍

**Pull request merged**